adafruit_74hc595.py

```diff
@@ -124,37 +134,41 @@ class ShiftRegister74HC595:
     and indicate the number of shift registers being used
     """
 
+    _device: spi_device.SPIDevice
+    _number_of_shift_registers: int
+    _gpio: bytes
```

Comment:
It looks like this gets initialized to `byterarray`.  In truth, the most widest applicable type for this is `ReadableBuffer`, which can be imported from `circuitpython_typing`.  To add that module, just add `adafruit-circuitpython-typing` to `requirements.txt`.  This signify that any buffer that can do reads (index read) can be used.

Reply:
What about ``val`` in the gpio setter? Also a ``ReadableBuffer``, or would you prefer ``bytearray`` like the constructor?


Currently:
```python
@gpio.setter
def gpio(self, val: bytes) -> None:
    self._gpio = val

    with self._device as spi:
        # pylint: disable=no-member
        spi.write(self._gpio)

Reply:
Yeah, we should you `ReadableBuffer` for this, `val` in the `gpio` setter, and the return of the `gpio` getter.